Program.cs

```diff
@@ -1,8 +1,43 @@
 ﻿class FizzBuzz
 {
-    static void Main(string[] args)
+    private static void mapOverIntegerRange(int start, int end, Action<int> action)
     {
-        Console.WriteLine("Hello, World!");
+        for (int index = start; index <= end; index++)
+        {
+            action(index);
+        }
+    }
+
+    private static bool divisibleBy(int number, int divisior)
+    {
+        return number % divisior == 0;
     }
-}
 
+    private static void responseToNumber(int number)
+    {
+        if (divisibleBy(number, 15)) // checking for divisibilty by both 3 and 5 at the same time
+        {
+            Console.WriteLine("FizzBuzz");
+        }

```

Comment:
That is interesting! I haven't initially thought about how non-prime numbers cannot use this trick, but it's true, it could lead to issues down the line. I modified it to a slightly less efficient '&&' statement that separately checks for both, removing the need for the comment.

Reply:
Great stuff (also I just noticed that using * for multiplying has made my github comment full of random italics 😅)